src/pilot/assign_role.py

```diff
@@ -262,32 +262,20 @@ def define_target_raid_config(role, drac_client):
         'logical_disks': logical_disks} if logical_disks is not None else None
 
 
-def is_raid_controller(fqdd):
-    return fqdd.startswith('RAID.')
-
-
-def is_boss_controller(fqdd, drac_client):
-    disk_controllers = drac_client.list_raid_controllers()
-    boss_raid_controller = [
-        c.id for c in disk_controllers if c.model.startswith('BOSS')]
-    if boss_raid_controller and fqdd == boss_raid_controller[0]:
-        return True
-
-
 def get_raid_controller_id(drac_client):
     disk_ctrls = drac_client.list_raid_controllers()
 
     raid_controller_ids = [
-        c.id for c in disk_ctrls if is_raid_controller(c.id)]
+        c.id for c in disk_ctrls if drac_client.is_raid_controller(c.id)]
     boss_controller_ids = [
-        c.id for c in disk_ctrls if is_boss_controller(c.id, drac_client)
+        c.id for c in disk_ctrls if drac_client.is_boss_controller(c.id)
     ]
 
     if boss_controller_ids:
         raid_controller_ids.append(boss_controller_ids[0])
     number_raid_controllers = len(raid_controller_ids)
 
-    if number_raid_controllers == 1:
+    if number_raid_controllers >= 1:
```

Comment:
@cdearborn  - Need discussion on this condition. We decided to go with this approach because we are getting more than 1 controllers always for any node.